src/components/PokemonTable.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import MUIDataTable from "mui-datatables";
+
+const renderTypes = types => types.join(" ");
+
+const shouldExcludePokemon = (pokemonTypes, filterTypes) => {
+  const result = filterTypes.every(val => pokemonTypes.includes(val));
+  return !result;
+};
+
+const getPokemonTypes = pokemon => [
+  ...new Set(pokemon.map(mon => [...mon.type, ...mon.weaknesses]).flat())
+];
+
+const createPokemonTableColumns = pokemon => {
+  const pokemonTypes = getPokemonTypes(pokemon);
+  return [
+    {
+      name: "name",
+      label: "Name",
+      options: {
+        filter: false,
+        sort: true
+      }
+    },
+    {
+      name: "num",
+      label: "Num",
```

Comment:
the mui-datatable lib API is pretty bad and honestly might have been more trouble then its worth. in this case `name` is actually used as a `key`.

instead of mapping all the pokemon to arrays of [name,num,type,weakness] you can pass in an array of objects and do a key lookup for values, you probably tried this but capitalized name so the lookup failed.